### PR TITLE
documents: complete the data conversion

### PR DESCRIPTION
**Pull request by reropag**
* Implements transformation from Marc21 to JSON RERO ILS  for:
    * intendedAudience (L26).
    * summary (L27).
    * contentMediaCarrier (L30).
    * original_language (L31).
    * note (L35).
    * originalTitle (L36).
    * adminMetadata (L34, L37, 45).
    * classification (L38).
    * sequence_numbering (L39).
    * dissertation (L40).
    * credits from (L41).
    * supplementaryContent from 504 (L42).
    * acquisitionTerms (L43).
    * tableOfContents (L44).
    * genreForm (L49).
    * subjects (L50).
    * subjects_imported (L53).
    * genreForm_imported (L54).

Co-Authored-by: Gianni Pante gianni.pante@rero.ch

## Why are you opening this PR?

- Which task/US does it implement?
- Which issue does it fix?

## Dependencies

My PR depends on the following `rero-ils-ui`'s PR(s):

* rero/rero-ils-ui#<xx>

## How to test?

- What command should I have to run to test your PR?
- What should I test through the UI?

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

**Review by BadrAly — APPROVED** [review]

File: rero_ils/dojson/utils.py [hunk 1]
```diff
@@ -222,6 +339,16 @@ def get_field_items(value):
         return utils.iteritems(value)
 
 
+def build_string_from_subfields(value, subfield_selection, separator):
+    """Build a string parsing the selected subfields in order."""
+    items = get_field_items(value)
+    string_part = []
+    for blob_key, blob_value in items:
+        if blob_key in subfield_selection:
+            string_part.append(blob_value)
```
Comment (zannkukai) [inline rero_ils/dojson/utils.py:348]: maybe use a `dict comprehension` ? 
```
parts = [ value for key, value in items if key in subfield_selection]
return separator.join(parts)
```

File: rero_ils/dojson/utils.py [hunk 2]
```diff
@@ -222,6 +339,16 @@ def get_field_items(value):
         return utils.iteritems(value)
 
 
+def build_string_from_subfields(value, subfield_selection, separator):
```
Comment (zannkukai) [inline rero_ils/dojson/utils.py:342]: interesting to use a default value for `separator` ?

File: rero_ils/dojson/utils.py [hunk 3]
```diff
@@ -837,27 +968,77 @@ def do(self, blob, ignore_missing=True, exception_handlers=None):
...
+                if note:
+                    notes_from_019_and_351.append(note[3:])
+
+            fields_351 = self.get_fields(tag='351')
+            note = ''
+            for field_351 in fields_351:
+                for subfield_c in self.get_subfields(field_351, 'c'):
+                    note += ' | ' + subfield_c
+                if note:
+                    notes_from_019_and_351.append(note[3:])
```
Comment (zannkukai) [inline rero_ils/dojson/utils.py:1019]:
```python
notes_from_019_and_351.append(' | '.join(self.get_subfields(field_351, 'c')))
```

**Review by lauren-d — APPROVED** [review]

**Review by iGormilhit — APPROVED** [review]
Commit message approved.

File: rero_ils/dojson/utils.py (hunk 3 above)
Comment (reropag) [reply rero_ils/dojson/utils.py:1019]: N'y a-t-il pas le risque d'ajouter une chaine vide en cas d'absence de $c?
Faut-il dans ce cas l'écrire sous cette forme?
note = ' | '.join(self.get_subfields(field_351, 'c'))
if note:
   notes_from_019_and_351.append(note)

File: rero_ils/dojson/utils.py (hunk 2 above)
Comment (reropag) [reply rero_ils/dojson/utils.py:342]: Bonne idée!
Je vais ajouter un `separator`par default.

File: rero_ils/dojson/utils.py (hunk 1 above)
Comment (reropag) [reply rero_ils/dojson/utils.py:348]: Bonne idée!

**Review by zannkukai — APPROVED** [review]

**Review by benerken — APPROVED** [review]

**Pull request merged**